ovnnbimp_test.go

```diff
@@ -475,4 +475,73 @@ func TestLoadBalancer(t *testing.T) {
 		t.Fatalf("error: lb deletion not done, total:%v", len(lb))
 	}
 	t.Logf("LB deletion done")
+	// test adding lb to lr and lswitch
+	ocmd, err = ovndbapi.LBAdd("lb2", "192.168.0.20:80", "tcp", []string{"10.0.0.21:80", "10.0.0.22:80"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ovndbapi.Execute(ocmd)
+	if err != nil {
+		t.Fatalf("Adding LB OVN failed with err %v", err)
+	}
+	t.Logf("Adding LB lb2 to LSwitches %s", LSW)
+	ocmd, err = ovndbapi.LSLBAdd(LSW, "lb2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ovndbapi.Execute(ocmd)
+	if err != nil {
+		t.Fatalf("Adding LB lb2 to LSwitch failed with err %v", err)
+	}
+	t.Logf("Adding LB lb2 to LSwitch %s Done", LSW)
+	t.Logf("List LB for LSwitch %s", LSW)
+	t.Logf("Deleting LB lb2 from LSwitch %s", LSW)
+	ocmd, err = ovndbapi.LSLBDel(LSW, "lb2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ovndbapi.Execute(ocmd)
+	if err != nil {
+		t.Fatalf("Deleting LB lb2 from LSwitch failed with err %v", err)
+	}
+	t.Logf("Deleting LB lb2 from LSwitch %s Done", LSW)
+	// Add LB to router
+	t.Logf("Adding LB to LRouter %s", LR)
+	ocmd, err = ovndbapi.LRLBAdd(LR, "lb2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ovndbapi.Execute(ocmd)
+	if err != nil {
+		t.Fatalf("Adding LB lb2 to LRouter failed with err %v", err)
+	}
+	t.Logf("Adding LB lb2 to LRouter %s Done", LR)
```

Comment:
List it to make sure it is added successfully.

Reply:
sure. Will do.

Reply:
@hzhou8 List will not show because we need to add update operations for cache else cache only updates insert and delete data. Do you want me to handle that in same PR? I already have mentioned this to @vtolstov and we can use different PR for updates!

Reply:
Any change to the DB should have been reflected back to the cache already, otherwise, existing tests should fail. For example: https://github.com/eBay/go-ovn/blob/master/dhcp_options_test.go#L101

If there is really a bug, this feature is anyway broken, so merging the PR doesn't help. We should create a new PR to fix the bug and then we can merge this PR. In any case, the test case for this PR should ensure the correctness of this feature, i.e. we should be able to immediately read whatever changes we made to DB.

Reply:
@hzhou8  . The problem was lrouter was getting deleted in the router test cases and hence listing lb from lr was failing when I added more debug. It's ok. Have addressed the same and have added two seperate test cases for lbls and lblr along with assert verification. PTAL.